internal/app/machined/pkg/system/runner/process/process.go

```diff
@@ -78,27 +88,75 @@ func (p *processRunner) Close() error {
 }
 
 type commandWrapper struct {
-	cmd              *exec.Cmd
+	launcher         *cap.Launcher
+	ctty             optional.Optional[int]
+	stdin            uintptr
+	stdout           uintptr
+	stderr           uintptr
 	afterStart       func()
 	afterTermination func() error
 }
 
 //nolint:gocyclo
 func (p *processRunner) build() (commandWrapper, error) {
-	args := []string{
-		fmt.Sprintf("-name=%s", p.args.ID),
-		fmt.Sprintf("-dropped-caps=%s", strings.Join(p.opts.DroppedCapabilities, ",")),
-		fmt.Sprintf("-cgroup-path=%s", cgroup.Path(p.opts.CgroupPath)),
-		fmt.Sprintf("-oom-score=%d", p.opts.OOMScoreAdj),
-		fmt.Sprintf("-uid=%d", p.opts.UID),
+	wrapper := commandWrapper{}
+
+	env := slices.Concat([]string{"PATH=" + constants.PATH}, p.opts.Env, os.Environ())
+	launcher := cap.NewLauncher(p.args.ProcessArgs[0], p.args.ProcessArgs[1:], env)
+
+	if p.opts.UID > 0 {
+		launcher.SetUID(int(p.opts.UID))
+	}
+
+	droppedCaps := strings.Join(p.opts.DroppedCapabilities, ",")
+
+	prop, err := krnl.ReadParam(&kernel.Param{Key: "proc.sys.kernel.kexec_load_disabled"})
+	if v := strings.TrimSpace(string(prop)); err == nil && v != "0" {
+		log.Printf("kernel.kexec_load_disabled is %s, skipping dropping capabilities", v)
+	} else if droppedCaps != "" {
+		caps := strings.Split(droppedCaps, ",")
+		dropCaps := xslices.Map(caps, func(c string) cap.Value {
+			capability, capErr := cap.FromName(c)
+			if capErr != nil {
+				fmt.Printf("failed to parse capability: %s", capErr)
+			}
+
+			return capability
+		})
+
+		// reduce capabilities and assign them to launcher
+		iab := cap.IABGetProc()
+		if err = iab.SetVector(cap.Bound, true, dropCaps...); err != nil {
+			return commandWrapper{}, fmt.Errorf("failed to set capabilities: %s", err)
```

Comment:
Always use `%w` for variables that implement or themselves are `error` type in `fmt.Errorf`. [Details](https://pkg.go.dev/fmt#Errorf).